--- a/hailo_apps/cpp/onnxrt_hailo_pipeline/CMakeLists.txt
+++ b/hailo_apps/cpp/onnxrt_hailo_pipeline/CMakeLists.txt
@@ -175,31 +175,10 @@ endif()
 # ============================================================
-# xtl + xtensor (header-only) - ExternalProject install (WORKING STYLE)
-# ============================================================
-# ---- ExternalProject for xtl/xtensor (header-only) ----
-include(ExternalProject)
-
-# A single install prefix where headers will land:
-set(EXTERNAL_PREFIX ${CMAKE_BINARY_DIR}/external)
-
-ExternalProject_Add(xtl-test
-    GIT_REPOSITORY https://github.com/xtensor-stack/xtl
-    GIT_TAG 0.7.7
-    CMAKE_ARGS -DCMAKE_INSTALL_PREFIX=${EXTERNAL_PREFIX}
-    -DCMAKE_POLICY_VERSION_MINIMUM=3.5
-    INSTALL_DIR ${EXTERNAL_PREFIX}
-)
-
-ExternalProject_Add(xtensor-test
-    DEPENDS xtl-test
-    GIT_REPOSITORY https://github.com/xtensor-stack/xtensor
-    GIT_TAG 0.24.6
-    CMAKE_ARGS
-        -DCMAKE_INSTALL_PREFIX=${EXTERNAL_PREFIX}
-        -Dxtl_DIR=${EXTERNAL_PREFIX}/share/cmake/xtl
-        -DCMAKE_POLICY_VERSION_MINIMUM=3.5
-    INSTALL_DIR ${EXTERNAL_PREFIX}
-)
-
-# xtensor include path (after external install)
-set(XTENSOR_INCLUDE_DIR ${EXTERNAL_PREFIX}/include)
+# xtl + xtensor — use sysroot install
+# ============================================================
+find_package(xtl REQUIRED)
+find_package(xtensor REQUIRED)

 # ============================================================
 # Sources
@@ -211,14 +190,11 @@ add_executable(${PROJECT_NAME}
     ../common/resources_manager.cpp
 )

-add_dependencies(${PROJECT_NAME} xtl-test xtensor-test)
-
 target_include_directories(${PROJECT_NAME} PRIVATE
     ${OpenCV_INCLUDE_DIRS}
     ${ONNXRUNTIME_INCLUDE_DIR}
     ${CMAKE_CURRENT_SOURCE_DIR}
-    ${XTENSOR_INCLUDE_DIR}
     ${CMAKE_CURRENT_SOURCE_DIR}/../common
-    ${CMAKE_BINARY_DIR}/external/include
 )
